src/nspawn/nspawn-mount.c

```diff
@@ -989,6 +989,20 @@ int mount_custom(
         return 0;
 }
 
+bool has_root_mount(CustomMount *mounts, size_t n) {
+        size_t i = 0;
+
+        for (i = 0; i < n; i++) {
+                CustomMount *m = mounts + i;
+
+                if (path_equal(m->destination, "/")) {
+                        return true;
+                }
+        }
+
+        return false;
+}
```

Comment:
hmm, why do you only check whether the root dir has any custom mount? why not check if it's actually an overlay one? can you elaborate?

Reply:
The documentation mentions that `--read-only` does not affect mounts made with `--tmpfs`, `--bind` and similar options. If we only check for overlays here, `--tmpfs` and `--bind` root mounts will still get remounted read-only which would not be according to the documented behaviour.